patterns/Factory-Design-Patten/README.md

```diff
@@ -0,0 +1,150 @@
+
+(Poorte, 2023)
+
+## Example demonstrating the Factory Method pattern in JavaScript:
+
+### Factory Function for Creating Foods
+
+The FoodFactory function serves as a factory for creating different types of foods, demonstrating the Factory Design Pattern. It contains a method createFood that takes a type as input and returns a corresponding food object. This approach abstracts the process of object creation, allowing the client code to create food objects without knowing the specific type being instantiated.
+
+```javascript
+// Factory Function for Creating Foods
+function FoodFactory() {
+    // Method to create different types of foods based on input type
+    this.createFood = function(type) {
+        let food;
+        
+        // Create different foods based on the type provided
+        if (type === "pizza") {
+            food = new Pizza();
+        } else if (type === "burger") {
+            food = new Burger();
+        } else if (type === "salad") {
+            food = new Salad();
+        }
+        
+        // Add common properties and methods to all created foods
+        food.type = type;
+        food.famousTo = function() {
+            console.log("The " + this.type + " is famous among: " + this.famousAudience);
+        }
+        
+        return food;
+    }
+}
+```
+### Pizza Class 
+
+The Pizza class represents a type of food created by the FoodFactory factory function. It encapsulates the properties and behavior specific to pizza objects. By being created through the FoodFactory, the Pizza class adheres to the Factory Design Pattern, where the factory method (createFood) is responsible for creating instances of concrete classes (Pizza) without exposing the instantiation logic.
+
+```javascript
+// Pizza Class
+function Pizza() {
+    this.famousAudience = "youngers!";
+}
+```
+
+### Burger Class 
+
+The Burger class represents another type of food created by the FoodFactory factory function. Similar to the Pizza class, it encapsulates the properties and behavior specific to burger objects. The instantiation of Burger objects through the FoodFactory aligns with the Factory Design Pattern, abstracting the object creation process and providing a common interface for creating different types of foods.
+
+```javascript
+// Burger Class
+function Burger() {
+    this.famousAudience = "Kids!";
+}
+```
+
+### Salad Class 
+
+The Salad class represents yet another type of food created by the FoodFactory factory function. It encapsulates the properties and behavior specific to salad objects. As with the Pizza and Burger classes, the instantiation of Salad objects through the FoodFactory adheres to the Factory Design Pattern, promoting encapsulation and decoupling by separating the client code from the object creation logic.
+
```

Comment:
While I appreciate your inclusion of these details in each food paragraph, I think you could have one paragraph to list the benefits attributable to all three classes, and remove the paragraphs for each food. That will cut down on unnecessary repetition.